packages/web/src/pages/upload-page/fields/CollectibleGatedDescription.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCallback } from 'react'
+
+import { collectiblesSelectors } from '@audius/common'
+import { Button, ButtonType } from '@audius/stems'
+import { useSelector } from 'react-redux'
+
+import { ReactComponent as IconExternalLink } from 'assets/img/iconExternalLink.svg'
+import { HelpCallout } from 'components/help-callout/HelpCallout'
+
+import styles from './CollectibleGatedDescription.module.css'
+
+const { getHasUnsupportedCollection } = collectiblesSelectors
+const messages = {
+  collectibleGated: 'Collectible Gated',
+  collectibleGatedSubtitle:
+    'Users who own a digital collectible matching your selection will have access to your track. Collectible gated content does not appear on trending or in user feeds.',
+  noCollectibles:
+    'No Collectibles found. To enable this option, link a wallet containing a collectible.',
+  compatibilityTitle: "Not seeing what you're looking for?",
+  compatibilitySubtitle:
+    'Unverified Solana NFT Collections are not compatible at this time.',
+  learnMore: 'Learn More'
+}
+
+const LEARN_MORE_URL =
+  'https://blog.audius.co/article/introducing-nft-collectible-gated-content'
+
+export const CollectibleGatedDescription = ({
+  hasCollectibles,
+  isUpload
+}: {
+  hasCollectibles: boolean
+  isUpload: boolean
+}) => {
+  const hasUnsupportedCollection = useSelector(getHasUnsupportedCollection)
+
+  const renderContent = useCallback(() => {
+    return hasUnsupportedCollection ? (
+      <div>
```

Comment:
nit but probably spans + maybe headings?

Reply:
oh als, can just make this a computed function, and then dont need to useCallback it since it is called in the render. useCallback only necessary when passing the function in as a prop to keep it stable